pkgs/package-overrides.nix

```
@@ -129,89 +197,120 @@ in
-        prev.mkExtension {
-          name = "mssql";
-          configureFlags = [
-            "--with-mssql=${pkgs.freetds}"
-          ];
-        }
+        prev.mkExtension
+          {
+            name = "mssql";
+            configureFlags = [
+              "--with-mssql=${pkgs.freetds}"
+            ];
+          }
       else
         null;
 
     mysql =
       if lib.versionOlder prev.php.version "7.0" then
-        prev.mkExtension {
-          name = "mysql";
-          internalDeps = [ prev.php.extensions.mysqlnd ];
-          configureFlags = [
-            "--with-mysql"
-            "--with-mysql-sock=/run/mysqld/mysqld.sock"
-          ];
-          postPatch = ''
-            # Fix mysql not being able to find headers.
-            ln -s $PWD/ext/ ext/mysql
-          '';
-        }
+        prev.mkExtension
+          {
+            name = "mysql";
+            internalDeps = [ prev.php.extensions.mysqlnd ];
+            configureFlags = [
+              "--with-mysql"
+              "--with-mysql-sock=/run/mysqld/mysqld.sock"
+            ];
+            postPatch = ''
+              # Fix mysql not being able to find headers.
+              ln -s $PWD/ext/ ext/mysql
+            '';
+          }
       else
         null;
 
     mysqli =
       if lib.versionOlder prev.php.version "7.0" then
-        prev.extensions.mysqli.overrideAttrs (attrs: {
-          # the --with-mysql-sock option didn't exist in php 5.6
-          NIX_CFLAGS_COMPILE = "-DPHP_MYSQL_UNIX_SOCK_ADDR=\"/run/mysqld/mysqld.sock\"";
-        })
+        prev.extensions.mysqli.overrideAttrs
+          (attrs: {
+            # the --with-mysql-sock option didn't exist in php 5.6
+            NIX_CFLAGS_COMPILE = "-DPHP_MYSQL_UNIX_SOCK_ADDR=\"/run/mysqld/mysqld.sock\"";
+          })
       else
         prev.extensions.mysqli;
 
-    mysqlnd =
-      if lib.versionOlder prev.php.version "7.1" then
-        prev.extensions.mysqlnd.overrideAttrs (attrs: {
-          postPatch = attrs.postPatch or "" + "\n" + ''
-            # Fix mysqlnd not being able to find headers.
-            ln -s $PWD/ext/ ext/mysqlnd
-          '';
-        })
-      else
-        prev.extensions.mysqlnd;
+    mysqlnd = prev.extensions.mysqlnd.overrideAttrs (attrs: {
+      patches =
+        let
+          upstreamPatches =
+            attrs.patches or [ ];
+
+          ourPatches =
+            lib.optionals (lib.versionOlder prev.php.version "7.4") [
+              # Introduced in https://github.com/NixOS/nixpkgs/commit/2e0d4a8b39a03a0db0c6c3622473d333a44d1ec1
+              ./patches/mysqlnd_fix_compression.patch
+            ];
+        in
+        ourPatches ++ upstreamPatches;
+
+      postPatch =
+        attrs.postPatch or ""
+          + lib.optionalString (lib.versionOlder prev.php.version "7.1") ''
+          # Fix mysqlnd not being able to find headers.
+          ln -s $PWD/ext/ ext/mysqlnd
+        '';
+
+      postConfigure =
+        attrs.postConfigure or ""
```

I would guess changing this to `preConfigure` should work. Fortunately, it seems to be the only code that was originally run in `postPhpize`.

```suggestion
      preConfigure =
        attrs.preConfigure or ""
```